webui-ros-joystick/nodejs/public/js/joystick.js

```diff
@@ -27,106 +24,112 @@ function removeJoystick() {
   }
 }
 
-function setView() {
-  removeJoystick();
-  joySize = 400;
-  if (joySize > $(window).height()) {
-    joySize = $(window).height();
-  }
-  if (joySize > $(window).width()) {
-    joySize = $(window).width();
-  }
-  max_joy_pos = joySize / 3;
-  createJoystick(
-    $(window).width() / 2,
-    $(window).height() / 2,
-    (joySize * 2) / 3
-  );
-}
-
 function repeat_velcmd(v_lin, v_ang) {
   moveAction(v_lin, v_ang);
   joystick_timeout = setTimeout(function () {
     repeat_velcmd(lin, ang);
   }, velocity_repeat_delay);
 }
 
-function createJoystick(x, y, d) {
-  joystickContainer = document.getElementById("joystick");
+function createJoystick(posX, posY, size) {
+
+  let joystickContainer = document.querySelector('.joystickContainer')
+  joystickContainer.style.top = (posY - size / 2) + "px";
+  joystickContainer.style.left = (posX - size / 2) + "px";
+
+  let ledSize = size / 10;
+  let joystickSize = size - ledSize;
+
+  let joystick = document.getElementById("joystick");
+  joystick.style.width = joystickSize / 2 + "px";
+  joystick.style.height = joystickSize / 2 + "px";
+  joystick.style.top = size / 4 + "px";
+  joystick.style.left = size / 4 + "px";
+
+  let outerCircle = document.getElementById('outerCircle')
+  outerCircle.style.width = joystickSize + "px";
+  outerCircle.style.height = joystickSize + "px";
+  outerCircle.style.top = ledSize / 2 + "px";
+  outerCircle.style.left = ledSize / 2 + "px";
+  outerCircle.style.background = JOYSTICK_COLOR;
+
+  let innerCircle = document.getElementById('innerCircle')
+  innerCircle.style.width = joystickSize / 2 + "px";
+  innerCircle.style.height = joystickSize / 2 + "px";
+  innerCircle.style.top = size / 4 + ledSize / 4 + "px";
+  innerCircle.style.left = size / 4 + ledSize / 4 + "px";
+  innerCircle.style.background = JOYSTICK_COLOR;
+
+  let ledCircle = document.getElementById("ledCircle");
+  ledCircle.style.width = size + "px";
+  ledCircle.style.height = size + "px";
+  ledCircle.style.top = "0px";
+  ledCircle.style.left = "0px";
+  if (joystick.hidden) {
+    ledCircle.style.backgroundColor = LED_RED_COLOR;
+  } 
 
   var options = {
-    zone: joystickContainer,
-    position: { left: x + "px", top: y + "px" },
-    mode: "static",
-    size: d,
-    color: "#222222",
-    restJoystick: true,
+    zone: joystick,
+    color: JOYSTICK_COLOR,
+    size: joystickSize,
+    fadeTime: 40,
   };
   manager = nipplejs.create(options);
+  manager.on("start", async function (evt, nipple) {
+    beginJoyPosX = nipple.position.x;
+    beginJoyPosY = nipple.position.y;
+    lin = 0;
+    ang = 0;
+
+    await new Promise(r => setTimeout(r, 30));
+    innerCircle.hidden = true;
+    outerCircle.hidden = true;
+    // move led
+    let ledPos = ledCircle.getBoundingClientRect();
+    let relativePosY = (nipple.position.y - size / 2) - ledPos.top;
+    let relativePosX = (nipple.position.x - size / 2) - ledPos.left;
```

Comment:
```suggestion
    let relativePosY = (stick.position.y - size / 2) - ledPos.top;
    let relativePosX = (stick.position.x - size / 2) - ledPos.left;
```

If it was an Easter egg attempt, then it is a good one. But can we be serious?

Reply:
In joystick library (nipplejs) joystick is often referred as a nipple. According to docs this is usually called data as it's contains basic data from joystick.

Reply:
If it is not our naming, than fine